include/devicetree/reserved.h

```diff
@@ -0,0 +1,137 @@
+/**
+ * @file
+ * @brief reserved-memory Devicetree macro public API header file.
+ */
+
+/*
+ * Copyright (c) 2021 Carlo Caione <ccaione@baylibre.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DEVICETREE_RESERVED_H_
+#define ZEPHYR_INCLUDE_DEVICETREE_RESERVED_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup devicetree-reserved Devicetree reserved-memory API
+ * @ingroup devicetree
+ * @{
+ */
+
+#include <linker/devicetree_reserved.h>
+
+/**
+ * @brief Get the pointer to the reserved-memory region
+ *
+ * Example devicetree fragment:
+ *
+ *     reserved: reserved-memory {
+ *         compatible = "reserved-memory";
+ *         ...
+ *         n: node {
+ *             reg = <0x42000000 0x1000>;
+ *         };
+ *      };
+ *
+ * Example usage:
+ *
+ *     DT_RESERVED_GET_ADDR(DT_NODELABEL(n)) // (uint8_t *) 0x42000000
+ *
+ * @param node_id node identifier
+ * @return pointer to the beginning of the reserved-memory region
+ */
+#define DT_RESERVED_GET_ADDR(node_id) _DT_RESERVED_START(node_id)
```

Comment:
I think this should be named something lke DT_RESERVED_GET_PTR or something so that it's clear it's not just an address, but a pointer.

Reply:
ACK.